### PR TITLE
Minor changes to Chapter 1

### PR DESCRIPTION
**Pull request by pavpanchekha**
Most of these changes are to the "Go Further" blocks that were recently added, plus some minor code things that unfortunately required changes across other chapters. There were a lot of small notes to discuss, though, that aren't included in here.

### CODE REVIEW

File: book/http.md [hunk 1]
```diff
@@ -857,55 +833,36 @@ make it so that, if your browser is started without a URL being given,
...
+header instead of always using `utf8`. Test it on a real site that
+doesn't use `utf8`, like `google.com`.
+
 *data:* Yet another scheme is *data*, which
 allows inlining HTML content into the URL itself. Try navigating to
 `data:text/html,Hello world!` in a real browser to see what happens. Add
 support for this scheme to your browser. The *data* scheme is especially
 convenient for making tests without having to put them in separate files.
 
-*Body tag:* Only show text in an HTML document if it is between
```
Comment (chrishtr) [inline book/http.md:866]: Too easy?

Comment (pavpanchekha) [reply book/http.md:866]: Students would solve it by splitting on `<body>`; there's a comment about not doing that but it feels poorly motivated. But more importantly if you implement this it then gets in the way at a later point, probably Ch 6 when you find you don't have any `<link>` elements for some reason.

File: book/http.md [hunk 2]
```diff
@@ -86,17 +85,15 @@ now talk to `example.org`.
...
-the architecure of the web, and coining the term REST to name this architecture.
-REST is defined in Fielding's [PhD thesis][rest-thesis], and is characterized
-by its key properties that allowed the web to grow in a decentralized way.
-Since then, the concept of a "RESTful API" has come into being, as a way to
-describe whether other networked APIs follow the definition of REST (though
-there does seem to be [some confusion][what-is-rest] about it).
+The URL syntax is defined in [RFC
+3987](https://tools.ietf.org/html/rfc3986), whose first author is Tim
+Berners-Lee---no surprise there! The second author is Roy Fielding, a
+key contributors to the design HTTP and also well-known for describing
```
Comment (chrishtr) [inline book/http.md:91]:
```suggestion
key contributor to the design HTTP and also well-known for describing
```

File: book/http.md [hunk 3]
```diff
@@ -86,17 +85,15 @@ now talk to `example.org`.
...
+the "REST" architecture of the web in his [PhD thesis][rest-thesis]
```
Comment (chrishtr) [inline book/http.md:92]:
```suggestion
the "REST" architecture of the web in his [PhD thesis][rest-thesis],
```

**Pull request merged**